### PR TITLE
RFC: in hacker news demo, use repo env argument for resources and executors

### PR DESCRIPTION
**Pull request by sryza**
This relies on a not-yet-existent API for providing deployment names to repositories at creation time. The idea is that the value for this would be determined at the deployment level. I.e. either in the dagster.yaml or in Dagster Cloud.

### CODE REVIEW

File: examples/hacker_news_assets/hacker_news_assets/repo.py [hunk 1]
```diff
@@ -1,47 +1,23 @@
...
-def hacker_news_assets_prod():
-    return [
-        prod_assets,
-        schedule_from_partitions(download_prod_job),
-        make_slack_on_failure_sensor(base_url="my_dagit_url.com"),
-        make_hn_tables_updated_sensor(activity_stats_prod_job),
-        make_hn_tables_updated_sensor(story_recommender_prod_job),
-    ]
-
+def hacker_news_assets(deployment_name: Optional[str]):
```
Comment (schrockn) [inline examples/hacker_news_assets/hacker_news_assets/repo.py:14]: How does this function even get the string?

Comment (sryza) [reply examples/hacker_news_assets/hacker_news_assets/repo.py:14]: The framework would provide it. Either by grabbing a value from dagster.yaml or something equivalent in Dagster Cloud.

Comment (schrockn) [reply examples/hacker_news_assets/hacker_news_assets/repo.py:14]: So this is a projected feature, rather than some feature I don't know about.

Comment (sryza) [reply examples/hacker_news_assets/hacker_news_assets/repo.py:14]: Yes yes correct. See PR description. Maybe should have proclaimed this more loudly in the title.

**Pull request closed without merging**